source/common/quic/quic_server_transport_socket_factory.cc

```diff
@@ -8,6 +8,7 @@
 #include "source/common/runtime/runtime_features.h"
 #include "source/common/tls/context_config_impl.h"
 #include "source/common/tls/server_context_impl.h"
+#include "source/common/tls/tls_certificate_selector_impl.h"
```

Comment:
This shouldn't be needed. You're probably missing an include in server_context_impl.h. Revert this and fix elsewhere.